src/common/lib/api.ts

```diff
@@ -33,64 +29,76 @@ export interface AccountInfoRes {
   name: string;
 }
 
-interface StatusRes {
+export interface StatusRes {
   configured: boolean;
   unlocked: boolean;
   currentAccountId: string;
 }
 
-interface UnlockRes {
+export interface UnlockRes {
   currentAccountId: string;
 }
 
 interface BlocklistRes {
   blocked: boolean;
 }
 
-export const getAccountInfo = () => msg.request<AccountInfoRes>("accountInfo");
-
-/**
- * stale-while-revalidate get account info
- * @param id - account id
- * @param callback - will be called first with cached (stale) data first, then with fresh data.
- */
-export const swrGetAccountInfo = async (
-  id: string,
-  callback?: (account: AccountInfo) => void
-): Promise<AccountInfo> => {
-  const accountsCache = await getAccountsCache();
-
-  return new Promise((resolve, reject) => {
-    if (accountsCache[id]) {
-      if (callback) callback(accountsCache[id]);
-      resolve(accountsCache[id]);
-    }
-
-    // Update account info with most recent data, save to cache.
-    getAccountInfo()
-      .then((response) => {
-        const { alias } = response.info;
-        const { balance: resBalance, currency } = response.balance;
-        const name = response.name;
-        const balance =
-          typeof resBalance === "number" ? resBalance : parseInt(resBalance); // TODO: handle amounts
-        const account = {
-          id,
-          name,
-          alias,
-          balance,
-          currency: currency || "BTC", // set default currency for every account
-        };
-        storeAccounts({
-          ...accountsCache,
-          [id]: account,
-        });
-        if (callback) callback(account);
-        return resolve(account);
-      })
-      .catch(reject);
-  });
+export const getAccountInfo = (): Promise<AccountInfoRes> =>
+  msg.request<AccountInfoRes>("accountInfo");
+
+export const getAccountInfoKey = (
+  id: string | null | undefined
+): string | null => (id ? `accountInfo/${id}` : null);
+
+// useSWR hook is expected to be called in every render, hence we expect parameter id null to skip fetching
+export const useAccountInfoCached = async (
+  id: string | null
+): Promise<AccountInfo | null> => {
+  const { data, error } = await useSWR<AccountInfoRes>(
+    getAccountInfoKey(id),
+    getAccountInfo
+  );
+
+  if (!data) return null;
+  // @Todo: how to handle this error in the calling function ?
+  if (error) throw error;
+
+  return buildAccountInfo(id, data);
 };
+
+// revalidation (mark the data as expired and trigger a refetch) for the resource
+export const mutateAccountInfo = async (
+  id: string | null
+): Promise<AccountInfo | null> => {
+  const data = await mutate<AccountInfoRes>(
```

Comment:
Just mutating the value should be enough - SWR will return the new value in the next render. You don't need to return buildAccountInfo() as the response.

Reply:
If I remove this the AccountMenu balance of the selcted account will not be updated after having a payment success of the "Receive" page. As far as I see it, there is no render in the AccountContext when the screen of the Receive page has received the account. For this reason the mutate being called, needs to refetch the account data and update the states in AccountContext

`
  const refetchAccountInfo = async () => {
    const accountInfo = await api.refetchAccountInfo(account?.id || null);
    if (accountInfo) {
      // eslint-disable-next-line
      setAccount(accountInfo as any);
      updateAccountBalance(accountInfo.balance, accountInfo.currency);
    }
  };
`

Reply:
It works now. This is resolved.